src/main/java/ch/njol/skript/bukkitutil/ItemUtils.java

```diff
@@ -361,4 +357,39 @@ public static boolean isGlass(Material material) {
 				return false;
 		}
 	}
+
+	/**
+	 * Applies a provided {@code Consumer} to the meta of the provided {@code ItemStack} and returns the updated {@code ItemStack} (with updated {@code ItemMeta}).
+	 * @param itemStack the item whose meta is to be changed using the provided Consumer
+	 * @param metaChanger a consumer to update the meta of the provided ItemStack
+	 * @param <T>
+	 * @return the updated item
+	 */
+	public static <T extends ItemMeta> ItemStack changeItemMeta(ItemStack itemStack, Consumer<T> metaChanger) {
```

Comment:
can't it just be `Consumer<ItemMeta>`? also, im not sure how often consumers are used to change the meta, so im not sure how useful this method actually is

Reply:
also 
```suggestion
	public static <T extends ItemMeta> ItemStack changeItemMeta(@NotNull ItemStack itemStack, @NotNull Consumer<T> metaChanger) {
```

Reply:
By using <T extends ItemMeta>, the generic type ensures that the provided Consumer operates on the specific subclass directly. This avoids casting entirely and allows the compiler to catch type mismatches at compile time, improving code reliability.